azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/action/ResourceCommonActions.java

```diff
@@ -0,0 +1,107 @@
+            final String description = AzureOperationBundle.title("common|resource.stop", s.name()).toString();
+            final boolean enabled = s instanceof Startable && ((Startable) s).isStoppable();
+            return new ActionView(new IView.Label.Static("Stop", "/icons/action/stop.svg", description), enabled);
+        };
+        am.registerAction(STOP, new Action<>(stop, stopActionView));
+
+        final Consumer<IAzureResource<?>> restart = s -> tm.runInBackground(AzureOperationBundle.title("common|resource.restart", s.name()), ((Startable) s)::restart);
+        final Function<IAzureResource<?>, IView.Label> restartActionView = s -> {
+            final String description = AzureOperationBundle.title("common|resource.restart", s.name()).toString();
+            final boolean enabled = s instanceof Startable && ((Startable) s).isRestartable();
+            return new ActionView(new IView.Label.Static("Restart", "/icons/action/restart.svg", description), enabled);
+        };
+        am.registerAction(RESTART, new Action<>(restart, restartActionView));
+
+        final Consumer<IAzureResource<?>> refresh = s -> tm.runInBackground(AzureOperationBundle.title("common|resource.refresh", s.name()), s::refreshChildren);
+        final Function<IAzureResource<?>, IView.Label> refreshActionView = s -> {
+            final String description = AzureOperationBundle.title("common|resource.refresh", s.name()).toString();
+            return new IView.Label.Static("Referesh", "/icons/action/refresh.svg", description);
+        };
+        am.registerAction(REFRESH, new Action<>(refresh, refreshActionView));
+
+        final Consumer<IAzureResource<?>> delete = s -> tm.runInBackground(AzureOperationBundle.title("common|resource.delete", s.name()), ((Removable) s)::remove);
+        final Function<IAzureResource<?>, IView.Label> deleteActionView = s -> {
+            final String description = AzureOperationBundle.title("common|resource.delete", s.name()).toString();
+            return new ActionView(new IView.Label.Static("Delete", "/icons/action/delete.svg", description), s instanceof Removable);
+        };
+        am.registerAction(DELETE, new Action<>(delete, deleteActionView));
+
+        final Consumer<IAzureResource<?>> openPortalUrl = s -> {
+            final Runnable runnable = () -> am.getAction(OPEN_URL).handle(s.portalUrl());
+            tm.runInBackground(AzureOperationBundle.title("common|resource.open_portal_url", s.name()), runnable);
+        };
+        final Function<IAzureResource<?>, IView.Label> openPortalUrlView = s -> {
+            final String description = AzureOperationBundle.title("common|resource.open_portal_url", s.name()).toString();
+            return new IView.Label.Static("Open In Portal", "/icons/action/portal.svg", description);
+        };
+        am.registerAction(OPEN_PORTAL_URL, new Action<>(openPortalUrl, openPortalUrlView));
```

Comment:
It looks there are a few different among these actions.
I suggest you a mind to extract these codes as a `yaml` configuration:
`
common.actions:
    start:
        name: "Start"
        label: "Start"
        progressMessage: "Start %s (%s)"
        icon: "/icon/action/start.png"
    stop:
        name: "Stop"
        label: "Stop"
        progressMessage: "Stop %s (%s)"
        icon: "/icon/action/stop.png"
springcloud.actions:
    ref: ${common.actions}
    deploy:
        name: "Deploy"
        label: "Deploy"
        progressMessage: "Deploy spring clound app (%s)"
        icon: "/icon/action/deploy.png"
        driverClass: com.xxx.xxx.Deployable
`